### DISCUSSION TITLE
Sorting files based on regex

### DISCUSSION THREAD

**Discussion by andrew-m-j** [SebastianMC/obsidian-custom-sort#173] [Q&A] [answered]
Hello, I've browsed the issues posted and docs, but I'm not sure if the plugin has the capability I'm looking for. I have rearranged some of my daily and weekly notes into the same folder. They have a different naming structure, but I'd like to sort them based on the date - i.e., a weekly note would be placed in order based on the first day of the week. My files already have these details in the names, like so:

- 2024-02-04
- 2024-02-08
- 2024-02-17
- 2024-W06 (02-05)
- 2024-W07 (02-12)
- 2024-W08 (02-19)

The first files here are daily notes, and the later are weekly notes. I'd like to end up with the following order:

- 2024-02-04
- 2024-W06 (02-05)
- 2024-02-08
- 2024-W07 (02-12)
- 2024-02-17
- 2024-W08 (02-19)

Is there a way to apply a sort based on the final dates, or only numerical values in these file names?

This seems close to the suffix-based sorting, except the closing parentheses on my weekly note throws it off. I did confirm a workaround as suggested [here in a somewhat similar question](https://github.com/SebastianMC/obsidian-custom-sort/issues/171) to use metadata with the following sort: `< a-z by-metadata: sort-date`, but I have to add that property to all of the files and this isn't as responsive. (Because they are made from a template, I can do that moving forward, but going back would be tedious.)

**Comment by SebastianMC** [comment]
Hi @andrew-m-j 

the sorting order which you think about would require deriving the date from the syntax `2024-W06 (02-05)` --> 2024-02-05. In other words some kind of pattern `YYYY-W?? (mm-dd)`.
The short answer is: not supported. There is no support for freeform date patterns in current version of the plugin.
At the same time this specific pattern can be easily added to the list of predefined patterns.

Please confirm the pattern to add, I guess the below would be ok:
`YYYY-Www (mm-dd)`
(and the week number acting only as a placeholder, part of the pattern, ignored when translating to date).

There is an oncoming release of the plugin and I can add the support for such a pattern, low effort.

At the same time the release date is unknown for now, due to time constraints on my side.

[Update]: created explicit ticket #178 not to forget to include this feature when the time comes

**Comment by SebastianMC** [comment, marked as answer]
Hi @andrew-m-j 

[plugin release 3.1.1](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/3.1.1) comes with the requested feature.
